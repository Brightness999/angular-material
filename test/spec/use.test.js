/**
 * @fileoverview Test the middleware interface.
 */

const Logality = require('../..');
const { stubLogality, cooldown } = require('../lib/tester.lib');

describe('Middleware interface', () => {
  stubLogality();
  test('Middleware will receive log context', () => {
    const logality = new Logality();

    let outputDone = false;
    logality.use((logContext) => {
      expect(logContext).toBeObject();
      expect(logContext).toMatchSnapshot();
      outputDone = true;
    });

    logality.get().log('hello world');
    expect(outputDone).toBeTrue();
  });
});
