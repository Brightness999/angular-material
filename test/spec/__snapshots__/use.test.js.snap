// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middleware interface Async Middleware will receive log context 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "Logality",
    },
    "source": Object {
      "file_name": "/test/spec/use.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello async world",
  "severity": 6,
}
`;

exports[`Middleware interface Middleware can mutate the Log Context 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "Logality",
    },
    "source": Object {
      "file_name": "/test/spec/use.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello bogus world",
  "middleware": true,
  "middleware2": true,
  "severity": 6,
}
`;

exports[`Middleware interface Middleware return value will be ignored 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "Logality",
    },
    "source": Object {
      "file_name": "/test/spec/use.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello bogus world",
  "severity": 6,
}
`;

exports[`Middleware interface Middleware will receive log context 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "Logality",
    },
    "source": Object {
      "file_name": "/test/spec/use.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
